File: src/frontend/domyra-app/src/app/models/widget.model.ts
import { Type } from "@angular/core";

export type WidgetKind = 'angular' | 'web-component' | 'html';

export interface Widget {
  alias: string;
  kind: WidgetKind;
  elementName?: string;          // for web components
  scriptUrl?: string;            // for web components
  html?: string;                 // for static widgets
  cols: number;
  rows: number;
}

export interface WidgetDefinition extends Widget {
  component: Type<unknown> | null; // for internal Angular widgets
}

export const widgetColumns: string[] = ['alias', 'state'];

File: src/backend/Domyra.Application/Features/Widgets/Models/WidgetModel.cs
using Domyra.Domain.Entities;
using Domyra.Domain.Enums;

namespace Domyra.Application.Features.Widgets.Models;

public sealed record WidgetModel(
    int Id,
    string Alias,
    WidgetState State,
    string? Kind,
    string? ElementName,
    string? ScriptUrl,
    string? Html,
    int Cols,
    int Rows
    )
{
    public static WidgetModel From(Widget widget) =>
        new(
            widget.Id,
            widget.Alias,
            widget.State,
            widget.Kind,
            widget.ElementName,
            widget.ScriptUrl,
            widget.Html,
            widget.Cols,
            widget.Rows
        );
}
